### PR TITLE
statistics: handle DDL events in the new PQ

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/55906

Problem Summary:

### What changed and how does it work?

This PR is based on https://github.com/pingcap/tidb/pull/56383

This PR handles most of the DDL events and tries to update the queue accordingly.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by Rustin170506** [comment]
@lance6716 @fzzf678 [`753f2e4` (#56435)](https://github.com/pingcap/tidb/pull/56435/commits/753f2e4104e519a103f394a4fe115040d8c38541) You can only focus on this commit. This PR is based on https://github.com/pingcap/tidb/pull/56383.(the first commit)

**Comment by Rustin170506** [comment]
/hold

Meger this after https://github.com/pingcap/tidb/pull/56383 is merged.

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2_ddl_handler.go [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+	case model.ActionExchangeTablePartition:
+		err = pq.handleExchangeTablePartitionEvent(sctx, event)
+	case model.ActionReorganizePartition:
+		err = pq.handleReorganizePartitionEvent(sctx, event)
+	case model.ActionAlterTablePartitioning:
+		err = pq.handleAlterTablePartitioningEvent(sctx, event)
+	case model.ActionRemovePartitioning:
+		err = pq.handleRemovePartitioningEvent(sctx, event)
+	default:
+		// Ignore other DDL events.
```
Comment (fzzf678) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2_ddl_handler.go:72]: Shall we consider print a log in the default branch

**Review by fzzf678 — APPROVED** [review]
queue_v2_ddl_handler.go rest LGTM

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2_ddl_handler.go [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+	return nil
+}
+
+func (pq *AnalysisPriorityQueueV2) handleReorganizePartitionEvent(
+	sctx sessionctx.Context,
+	event *notifier.SchemaChangeEvent,
+) error {
+	globalTableInfo,
+		_,
+		droppedPartitionInfo := event.GetReorganizePartitionInfo()
```
[the remark this answers is not in the document]
Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2_ddl_handler.go]: Updated.

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2_ddl_handler.go (hunk 1 above)
Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2_ddl_handler.go:72]: It is OK for now.

**Review by winoros — APPROVED** [review]

**Pull request merged**